packages/loader/container-loader/src/container.ts

```diff
@@ -103,10 +102,37 @@ interface ILocalSequencedClient extends ISequencedClient {
     shouldHaveLeft?: boolean;
 }
 
+export interface IContainerLoadOptions {
```

Comment:
It feels to me that code would get more readable if IContainerLoadOptions derives from IContainerConfig and 
all the args that are passed directly (resolvedUrl, resolvedUrl, docId) are simply added to IContainerConfig.
Basically you only one thing to pass at construction - IContainerLoadOptions, and all unique fields are used only during load, but sub-part of it (IContainerConfig) is captured for duration of container.

Reply:
We probably have options to simplify here, but for this change I'm primarily focused on preserving the options matrix 1:1 with the current capabilities through the headers.  Merging to `IContainerConfig` may or may not be beneficial, but it would seem to grow the support matrix (e.g. I don't know what "`pause`" should mean in `IContainerConfig`, since it's only intended to be used as a param to `container.load()`).

Reply:
I think I understand what you mean better now, though the problem is that in the `Container.load` flow some of these configs are mandatory whereas they are optional in the `createDetached` flow.  I will experiment to see if there are cleaner options.

Reply:
Latest changes move the args into the property bag which I do think reads easier, but establishing a derived relationship between the config and the load options seems messy since they're non-optional in the load flow but excluded in the `createDetached` flow.  We could potentially benefit from greater separation of the config (i.e. constructor takes `ILoadedContainerConfig | IDetachedContainerConfig`), where none of the configurations would be optional.  But if we want to go that way I'd prefer to do it as a separate PR.